tools/tf2openapi/types/tftensor.go

```diff
@@ -34,30 +37,80 @@ type TFDType int
 const (
 	// all the possible constants that can be JSON-ified according to
 	// https://www.tensorflow.org/tfx/serving/api_rest#json_mapping
-	DT_BOOL TFDType = iota
-	DT_B64_STRING
-	DT_STRING
-	DT_INT8
-	DT_UINT8
-	DT_INT16
-	DT_INT32
-	DT_UINT32
-	DT_INT64
-	DT_UINT64
-	DT_FLOAT
-	DT_DOUBLE
+	// along with a representation for B64 strings
+	DtBool TFDType = iota
+	DtB64String
+	DtString
+	DtInt8
+	DtUInt8
+	DtInt16
+	DtInt32
+	DtUInt32
+	DtInt64
+	DtUint64
+	DtFloat
+	DtDouble
 )
 
-func NewTFTensor(key string, tensorInfo pb.TensorInfo) TFTensor {
-	return TFTensor{}
+func NewTFTensor(key string, tensor pb.TensorInfo) TFTensor {
+	if tensor.TensorShape.UnknownRank {
+		return TFTensor{
+			Key:   key,
+			DType: NewTFDType(tensor.Dtype.String(), key),
+			Shape: nil,
+			Rank:  -1,
+		}
+	} else {
```

Comment:
You don't have to put this in an else. One way I think about logic:
```
if (exceptionalcase) {
return
}

if (otherexceptionalcase) {
return
}

return normalcase
```

Elses add additional complexity to be reasoned about. It's not that bad here, but it's a general tip to keep indentation low and avoid nesting ifs and complex logic.

Reply:
+1